### PR TITLE
Add Python Solution to CSES Longest Flight Route

### PR DESCRIPTION
**Pull request by MartinLwx**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

**Comment by SansPapyrus683** [comment]
does this follow the sol described in the module

**Comment by MartinLwx** [comment]
> does this follow the sol described in the module

It follows the sol described in the module. The slight difference is that I separated the processes of solving the topological sorting and updating the array for dynamic programming into two separate steps.

File: content/4_Gold/TopoSort.mdx [hunk 1]
```diff
@@ -927,6 +927,63 @@ public class Main {
 ```
 
 </JavaSection>
+<PySection>
+
+```py
+from collections import defaultdict, deque
+
+
+def main():
```
Comment (SansPapyrus683) [inline content/4_Gold/TopoSort.mdx:936]: why `main()`

Comment (MartinLwx) [reply content/4_Gold/TopoSort.mdx:936]: This is an optimization to speed up python code. The local variable lookup will be much faster compare to global variable lookup. Please refers to this [link](https://stackoverflow.com/questions/12590058/performance-with-global-variables-vs-local) to get more details.

Comment (SansPapyrus683) [reply content/4_Gold/TopoSort.mdx:936]: then add a doc comment explaining that

Comment (MartinLwx) [reply content/4_Gold/TopoSort.mdx:937]: Done!

File: content/4_Gold/TopoSort.mdx [hunk 2]
```diff
@@ -927,6 +927,64 @@ public class Main {
...
+
+	topo_order = []
+	queue = deque([i for i, v in enumerate(in_degree) if v == 0])
+	while queue:
+		cur_node = queue.popleft()
+		topo_order.append(cur_node)
+		for neigh_node in edge[cur_node]:
+			in_degree[neigh_node] -= 1
+			if in_degree[neigh_node] == 0:
+				queue.append(neigh_node)
```
Comment (SansPapyrus683) [inline content/4_Gold/TopoSort.mdx:958]: comment for this block

File: content/4_Gold/TopoSort.mdx [hunk 3]
```diff
@@ -927,6 +927,64 @@ public class Main {
...
+				dist[v] = dist[prev] + 1
+				parent[v] = prev
+
+	if dist[-1] == -float("inf"):
+		print("IMPOSSIBLE")
+	else:
+		print(dist[-1])
+		cursor = cities - 1
+		path = []
+		while parent[cursor] != -1:
```
Comment (SansPapyrus683) [inline content/4_Gold/TopoSort.mdx:977]: and this one

File: content/4_Gold/TopoSort.mdx (hunk 2 above)
Comment (MartinLwx) [reply content/4_Gold/TopoSort.mdx:959]: Completed

File: content/4_Gold/TopoSort.mdx (hunk 3 above)
Comment (MartinLwx) [reply content/4_Gold/TopoSort.mdx:982]: Completed

**Review by SansPapyrus683 — APPROVED** [review]

**Review by ryanchou-dev — APPROVED** [review]
hell yeah

**Pull request merged**